sql/dbschema.sql

```diff
@@ -0,0 +1,99 @@
+CREATE TABLE users (
+  userId INT PRIMARY KEY IDENTITY(1,1),
+  firstName VARCHAR(50) NOT NULL,
+  lastName VARCHAR(50) NOT NULL,
+  email VARCHAR(50) NOT NULL,
+  deleted INT DEFAULT 0 NOT NULL
```

Comment:
Why do we need `deleted`? What is the purpose of having `deleted` field?

Reply:
I thought it might be a safer solution to update deleted field from 0 to 1 when a user or task gets deleted. in this way we avoid delete statement for more security and also we can restore the deleted record if we want. in this case the select statement will be like this => 
SELECT * FROM users WHERE deleted = 0;

I can remove it if you think it's redundant.

Reply:
Do we want to have a created at field as well?

Reply:
good idea, I will add it.

Reply:
From the perspective of someone who writes the sql query based on the db setup, "created" and "deleted" make logical sense, but will not be returned as part of API response. With that, I'd vote to leave out